extensions/WSLExtension/Models/WslComputeSystem.cs

```diff
@@ -0,0 +1,348 @@
+                UpdateState(ComputeSystemState.Unknown);
+                return GetErrorResult(ex, "WSLTerminateError");
+            }
+        }).AsAsyncOperation();
+    }
+
+    public IAsyncOperation<ComputeSystemOperationResult> DeleteAsync(string options)
+    {
+        return Task.Run(() =>
+        {
+            try
+            {
+                UpdateState(ComputeSystemState.Deleting);
+                _wslManager.UnregisterDistribution(Id);
+                UpdateState(ComputeSystemState.Deleted);
+                return new ComputeSystemOperationResult();
+            }
+            catch (Exception ex)
+            {
+                _log.Error(ex, $"Unable to unregister Distribution: {Id}");
+                UpdateState(ComputeSystemState.Unknown);
+                return GetErrorResult(ex, "WSLUnregisterError");
+            }
+        }).AsAsyncOperation();
+    }
+
+    public IAsyncOperation<ComputeSystemThumbnailResult> GetComputeSystemThumbnailAsync(string options)
+    {
+        return Task.Run(async () =>
+        {
+            try
+            {
+                if (string.IsNullOrEmpty(_distribution.Base64StringLogo))
+                {
+                    // No logo for this distribution so we'll use its PackageFamily logo
+                    var packageFamilyLogo =
+                        await _packageHelper.GetPackageIconAsByteArrayAsync(_distribution.PackageFamilyName!);
+
+                    if (packageFamilyLogo == null)
+                    {
+                        // Couldn't find package family logo, so instead use the default WSL logo.
+                        var defaultLogoBase64 = await _packageHelper.GetBase64StringFromLogoPathAsync(DefaultWslLogoPath);
+                        return new ComputeSystemThumbnailResult(FromBase64String(defaultLogoBase64));
+                    }
+
+                    return new ComputeSystemThumbnailResult(packageFamilyLogo);
+                }
+
+                // This is a known distribution. Use the logo we have for this distribution defined in
+                // the DistributionDefinitions.yaml
+                return new ComputeSystemThumbnailResult(FromBase64String(_distribution.Base64StringLogo!));
+            }
+            catch (Exception ex)
+            {
+                _log.Error(ex, $"Unable to get thumbnail for Distribution: {Id}");
+                return new ComputeSystemThumbnailResult(ex, ex.Message, ex.Message);
+            }
+        }).AsAsyncOperation();
+    }
+
+    public IAsyncOperation<IEnumerable<ComputeSystemProperty>> GetComputeSystemPropertiesAsync(string options)
+    {
+        return Task.Run(() =>
+        {
+            try
+            {
+                var properties = new List<ComputeSystemProperty>();
+                if (_distribution.Version2 != null)
+                {
+                    // There are only 2 versions of WSL. Version 1 and version 2.

```

Comment:
Hmm although highly unlikely they will ever make a v3, I'll just update the _distribution object to just have a _distribution.Version property instead of _distribution.Version2. And use the integer value directly.